Apps/Src/RTOSPedals.c

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+
+#include "RTOSPedals.h"
+
+/**
+* @brief   Initialize both pedals to 0%
+* @param   None
+* @return  None
+*/ 
+static void Pedals_Values_Init(void){
+    accelerator_percentage = 0;
+    brake_percentage = 0;
+}
+
+/**
+* @brief   Read both pedal percentage values and set the global variables
+* @param   None
+* @return  None
+*/ 
+void Task_ReadPedals(void *p_arg){
+    Pedals_Values_Init();
+    
+    OS_ERR err;
+
+    while(1){
+        accelerator_percentage = Pedals_Read(ACCELERATOR);
+        brake_percentage = Pedals_Read(BRAKE);
+
+        // Post to semaphores (velocityChange, lightsChange)
+        OSSemPost (&VelocityChange_Sem4, OS_OPT_POST_ALL, &err);
+        OSSemPost (&LightsChange_Sem4, OS_OPT_POST_ALL, &err);
```

Comment:
If you're planning on just looping infinitely and posting to these semaphores every time, we may as well just have the velocity and lights threads run without blocking semaphores and with some delay in the loop that still makes them responsive.

This would also probably be a bit better just so a change doesn't sneak by that task when someone forgets to post to a semaphore in the future.

Reply:
The idea with these is to have it only post the semaphores when the pedal position changes